src/test/java/org/kohsuke/github/Github2faTest.java

```diff
@@ -0,0 +1,52 @@
+package org.kohsuke.github;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Test;
+public class Github2faTest  extends AbstractGitHubWireMockTest {
+	public String twoFactorAuthCodePrompt() {
+		System.out.print("Github 2 factor temp key: ");
+		// create a scanner so we can read the command-line input
+		BufferedReader buf = new BufferedReader(new InputStreamReader(System.in));
+		// TODO Auto-generated method stub
+		try {
+			return buf.readLine().trim();
+		} catch (IOException e) {
+			return null;
+		}
+	}
+	@Test
+	public void test2faToken() {
+		DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
+		String timestamp = dateFormat.format(new Date());
+		List<String> asList = Arrays.asList("repo", "gist", "write:packages", "read:packages", "delete:packages",
+				"user", "delete_repo");
+		String string = "Test2faTokenCreate-" + timestamp;// use time stamp to ensure the token creations do not collide with older tokens
+		GHAuthorization token=null;
+		try {
+			token = gitHub.createToken(asList, string, "this is a test token created by a unit test");
+		}catch (IOException ex) {
+			// under 2fa mode this exception is expected, and is necessary
+			// as the exception is called, GitHub will generate and send an OTP to the users SMS
+			// we will prompt at the command line for the users 2fa code 
+			try {
+				//token = gitHub.createTokenOtp(asList, string, "", twoFactorAuthCodePrompt());// prompt at command line for 2fa OTP code
```

Comment:
Ah, I see.  Yeah, it's a little harder than I first thought. 

Here's my suggestion: 
* Add checks that right information is in the `GHAuthorization` that came back from the server.
* Add a thread sleep of 30 seconds after your call to `createTokenOtp()`. 
* Add actions and checks that verify things are as they should *as though the user responded to the SMS* 
* Run the test with `takeSnapshot` with yourself as the user.  
* Respond to the SMS while the test is waiting. 
* Let the test finish successfully
* Comment out the thread sleep with info explaining this is where a person would get an SMS message and respond
* Add `assumeFalse("Test only valid when not proxying", mockGitHub.isUseProxy());` to the top of your method. 
* Clean up the permissions your account and checkin all the files. 

This will help people understand what this method is for and will ensure if someone goes to change it they have some idea of what needs to be done. 

Cool?

Reply:
ok, cool, so the takeSnapshot will record the user interaction. I would then bake in the OTP code during the capture that I get, into the unit test, to match the header data. That way it can be run in mock. Is that right?